### PR TITLE
Backend/get param validation for space, topic and resource

### PR DESCRIPTION
**Pull request by furkanakkurt1335**
I've implemented parameter validations for `space`, `topic` and `resource` _GET_ functions, where they are needed.

Related issue: #381

### CODE REVIEW

File: app/server/src/routes/topic/topic.validate.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
   switch (method) {
-    case "create-topic":
+    case "create-topic": {
       return [
         body("name", "name doesn't exist").exists(),
         body("space_id", "space_id doesn't exist").exists(),
       ];
+    }
+    case "get-topic": {
+      return [param("id", "id does not exist").exists()];
```
Comment (ersoykadir) [inline app/server/src/routes/topic/topic.validate.js:12]: can you add ismongoid check to these

File: app/server/src/routes/topic/topic.validate.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+    }
+    case "get-topic-resources": {
+      return [param("id", "id does not exist").exists()];
```
Comment (ersoykadir) [inline app/server/src/routes/topic/topic.validate.js:15]: can you add ismongoid check to these

File: app/server/src/routes/space/space.validate.js [hunk 3]
```diff
@@ -10,5 +10,8 @@ exports.validate = (method) => {
         body("image", "image does not exist").exists(),
       ];
     }
+    case "getSpaceDetail": {
+      return [param("id", "id does not exist").exists()];
```
Comment (ersoykadir) [inline app/server/src/routes/space/space.validate.js:14]: can you add ismongoid check to these

Comment (ersoykadir) [reply app/server/src/routes/space/space.validate.js:14]: Can you also add case for get all discussions

File: app/server/src/routes/resource/resource.validate.js [hunk 4]
```diff
@@ -21,5 +21,6 @@ exports.validate = (method) => {
         body("resource_id", "resource_id doesn't exist or broken").exists().isMongoId(),
         body("rating", "rating doesn't exist").exists(),
       ];
+    }
```
Comment (ersoykadir) [inline app/server/src/routes/resource/resource.validate.js:24]: You forgot to add case for get-resource I suppose?

**Review by ersoykadir — CHANGES_REQUESTED** [review]
Good to go after adjustments.

File: app/server/src/routes/topic/topic.validate.js (hunk 1 above)
Comment (furkanakkurt1335) [reply app/server/src/routes/topic/topic.validate.js:12]: Done!

File: app/server/src/routes/topic/topic.validate.js (hunk 2 above)
Comment (furkanakkurt1335) [reply app/server/src/routes/topic/topic.validate.js:15]: Done!

File: app/server/src/routes/resource/resource.validate.js (hunk 4 above)
Comment (furkanakkurt1335) [reply app/server/src/routes/resource/resource.validate.js:24]: Yep, done!

File: app/server/src/routes/space/space.validate.js (hunk 3 above)
Comment (furkanakkurt1335) [reply app/server/src/routes/space/space.validate.js]: Done!

**Review by ersoykadir — APPROVED** [review]
LGTM!

**Pull request merged**